src/mongocxx/database.hpp

```diff
@@ -248,6 +248,8 @@ class MONGOCXX_API database {
 
     class MONGOCXX_PRIVATE impl;
     std::unique_ptr<impl> _impl;
+    const impl& get_impl() const;
+    impl* get_impl();
```

Comment:
@samantharitter suggested the return types for these functions. I guessed she made the return type of the const function a reference so that we wouldn't need to make both the pointer and the contents of the pointer const, but I wasn't totally sure. Samantha, would you mind explaining your reasoning?